Translation of Apollo Official Chinese Document(s) (#4322)
apolloconfig/apollo: docs/en/usage/dotnet-sdk-user-guide.md

   * If no configuration is found, we will try to load the configuration from the cluster specified by `idc`.
   * If nothing is found, we will load from the default cluster (`default`)

2. If only `Apollo.Cluster` is specified.
   * We will first try to load the configuration from the cluster specified by `Apollo.Cluster`
   * If not found, it will be loaded from the default cluster (`default`)

3. If only `idc` is specified.
   * We will first try to load the configuration from the cluster specified by `idc`
   * If not found, it will be loaded from the default cluster (`default`)

4. If neither `Apollo.Cluster` nor `idc` is specified.
   * It will load the configuration from the default cluster (`default`)

# II. DLL references

Net client project address is located at: [https://github.com/ctripcorp/apollo.net](https://github.com/ctripcorp/apollo.net).

Download the project locally, switch to the `Release` configuration, compile the Solution and it will generate `Framework.Apollo.Client.dll` in `apollo.net\Apollo\bin\Release`.

Just reference `Framework.Apollo.Client.dll` in your application.

> .Net Core, you can refer to [dotnet-core](https://github.com/ctripcorp/apollo.net/tree/dotnet-core) and [nuget repository](https://www.nuget. org/packages?q=Com.Ctrip.Framework.Apollo)

Reviewer: ```suggestion
.Net Core, you can refer to [dotnet-core](https://github.com/ctripcorp/apollo.net/tree/dotnet-core) and [nuget repository](https://www.nuget.org/packages?q=Com.Ctrip.Framework.Apollo)
```

# III. Client-side usage

## 3.1 Get the configuration of the default namespace (application)

```c#
Config config = ConfigService.GetAppConfig(); //config instance is singleton for each namespace and is never null
string someKey = "someKeyFromDefaultNamespace";
string someDefaultValue = "someDefaultValueForTheKey";
string value = config.GetProperty(someKey, someDefaultValue);
```

> GetProperty **can get the real-time latest configuration value corresponding to someKey by the above **config.

Reviewer: ```suggestion
With the above **config.getProperty** you can get the real-time latest configuration value corresponding to someKey.
```

   * 4. The client will cache a copy of the configuration obtained from the server in the local file system. In case of service unavailability or network failure, the configuration can still be restored locally
5. The application can get the latest configuration from the Apollo client, subscribe to configuration update notifications

# V. Local Development Mode

Apollo client also supports local development mode, which is mainly used when the development environment cannot connect to Apollo server, such as doing related function development on cruise ships or airplanes.

In local development mode, Apollo will only read configuration information from local files, not from Apollo server.

You can enable Apollo local development mode by following the steps below.

## 5.1 Modify the environment

Modify the C:\opt\settings\server.properties file to set the env to Local: 

```properties
env=Local
```

## 5.2 Preparing local configuration files

In local development mode, Apollo client will read files from local, so we need to prepare the configuration file beforehand.

### 5.2.1 Local configuration directory

> The local configuration directory is located at: `C:\opt\data\{_appId_}\config-cache`.

Reviewer: ```suggestion
The local configuration directory is located at: C:\opt\data\{_appId_}\config-cache.
```
